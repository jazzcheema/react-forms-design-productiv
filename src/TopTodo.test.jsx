import {describe, it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo.jsx";



describe("TopTodo component ", function () {

    const testTodos = [
    {
        title: "Test Title1",
        description: "Test Description1",
        priority: "1"
    }
    ,
    {   title: "Test Title2",
        description: "Test Description2",
        priority: "2"
    }
    ,
    {   title: "Test Title3",
        description: "Test Description3",
        priority: "3"}
    ]

    it("Renders without crashing", function () {
      render(<TopTodo todos={testTodos}/>);
    });

    it("Matches snapshot", function(){
        const {container} = render(<TopTodo todos={testTodos} />);
        expect(container).toMatchSnapshot();
    })

    it("Renders top priority todo for list of todos", function(){
        const {container} = render(<TopTodo todos={testTodos}/>);
        const title = container.getElementsByTagName('b')[0];
        expect(title.textContent).toEqual("Test Title1");
    })

});