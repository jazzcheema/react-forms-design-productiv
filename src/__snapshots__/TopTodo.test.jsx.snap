// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TopTodo component  > Matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        Test Title1
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        Test Description1
      </small>
    </div>
  </div>
</div>
`;
